meta-balena-common/recipes-support/resin-init/resin-init-flasher/balena-init-flasher-efi

```diff
@@ -48,15 +48,42 @@ secureboot_setup() {
 
             if [ "${SETUPMODE_VAR}" -eq "1" ]; then
                 info "Secure boot setup mode detected - programming keys"
+
+                DB_FILES="db-loader db-kernel"
+                DB_EXTRA_FILE=""
+
+                # If PCIe firmware or UEFI drivers are executed and measured into PCR2, whitelist them
+                TPM_MEASUREMENTS_FILE="/sys/kernel/security/tpm0/binary_bios_measurements"
+                if [ -f "${TPM_MEASUREMENTS_FILE}" ]; then
+                    PCR2_HASHES=$(tpm2_eventlog "${TPM_MEASUREMENTS_FILE}" \
+                                  | awk -F': ' '
+                                      $1 ~ /PCRIndex/ { pcr = $2 }
+                                      $1 ~ /EventType/ { event = $2 }
+                                      $1 ~ /AlgorithmId/ && event != "EV_SEPARATOR" && pcr == 2 && $2 == "sha256" { digestFlag = 1 }
+                                      $1 ~ /Digest/ && digestFlag { print $2; digestFlag = 0 }' \
+                                  | sed -e 's,",,g' \
+                                  | sort -u)
+
+                    if [ -n "${PCR2_HASHES}" ]; then
```

Comment:
As we discussed previously, I think this can be made opt-in and I think it looks good.